aea/configurations/data_types.py

```diff
@@ -321,6 +321,21 @@ def from_str(cls, public_id_string: str) -> "PublicId":
         version = match.group(3)[1:] if ":" in public_id_string else None
         return PublicId(username, package_name, version)
 
+    @classmethod
+    def try_from_str(cls, public_id_string: str) -> Optional["PublicId"]:
+        """
+        Safely try to get public id from string.
+
+        :param public_id_string: the public id in string format.
+        :return: the public id object or None
+        """
+        result: Optional[PublicId] = None
+        try:
+            result = cls.from_str(public_id_string)
+        except ValueError:
+            pass
+        return result
```

Comment:
(minor) A slightly less verbose alternative:
```suggestion
import contextlib

        result: Optional[PublicId] = None
        with contextlib.suppress(ValueError):
            result = cls.from_str(public_id_string)
        return result
```